Dockerfile

```diff
@@ -496,7 +496,7 @@ USER root
 RUN dnf -y install golang
 
 ENV GO123VER="1.23.7"
-ENV GO122VER="1.22.12"
+ENV GO124VER="1.24.1"
```

Comment:
I'm posting this comment here for the lack of a better place to post a comment and request changes

I synced with @bcressey offline about the FIPS support in Go 1.24. Given that using the native Go support requires changes in how we package Bottlerocket binaries, for this release we **must** keep the patch to support FIPS. We have to track migrating to the new native FIPS support for Bottlerocket binaries since our current approach through the patch won't be supported going forward:

> The previous, unsupported mechanism to use the BoringCrypto module for certain FIPS 140-3 approved algorithms is currently still available, but it is meant to be removed and replaced with the mechanism described in this page in a future release.